dsalgo/Search.py

```diff
@@ -0,0 +1,87 @@
+import sys
+
+class Search:
+
+	def __new__(self, array, algorithm, element):
+
+		'''
+		args:
+	      :array: (list) : a python list
+	      :algorithm: (str): Searching algorithm type
+	                        Values supported are:
+	                        1. linear
+	                        2. binary
+
+	      :element: Element whose index has to be found
+	                if present  
+	    
+	    return:
+	        Index of the element if found else returns -1
+		'''
+
+
+		self.array = array
+		self.algorithm = algorithm
+		self.element = element
+
+		if self.algorithm == 'linear':
+			return linear_search(self.array, self.element)
+
+		elif self.algorithm == 'binary':
+			return binary_search(self.array, 0, len(self.array)-1, self.element)
+
+		else:
+			stderr.write("Error: unsupported searching algorithm passed!")
+
+
+def linear_search(array, element):
+	
+	'''
+	args:
+      :array: (list) : a python list
+      :element: Element whose index has to be found
+                if present  
+    
+    return:
+        Index of the element if found else returns -1
+	'''
+
+	for index, value in enumerate(array):
+		if value == element:
+			return index
+
+	return -1
+
+
+
+def binary_search(array, l, r, element): 
```

Comment:
code for binary search is wrong. you need to pass sorted list here.

Reply:
Tell me how you plan to design the library. 
Do you want to add sorting to be an option ? Like should I sort it by default, for binary search or do you want me to throw an error if the array given is not sorted?

Because if we sort it, the index positions would change. Do you want me to output the index value or just return true or false corresponding to whether the element is present or not?

@codesankalp

Reply:
I want you to show a warning when array is not sorted for binary search. But search the element in sorted one @GSNCodes.